libraries/api-extractor-model/src/items/ApiDeclaredItem.ts

```diff
@@ -74,6 +79,20 @@ export class ApiDeclaredItem extends ApiDocumentedItem {
     return this._excerptTokens;
   }
 
+  /**
+   * The source file where the API item is declared. Is `undefined` if the
+   * source file could not be determined.
+   *
+   * @remarks
+   * The source file path is relative with respect to the `<projectFolder>` token as defined in
+   * the `api-extractor.json` config. It points to the `.ts` source file where the API item was
+   * originally declared. However, in some cases, if source map resolution fails, it falls back
+   * to pointing to the `.d.ts` file instead.
+   */
+  public get sourceFile(): string | undefined {
```

Comment:
@zelliott I wrote some more thoughts about this in https://github.com/microsoft/rushstack/issues/895#issuecomment-1223348529

We can postpone most of it until a later PR.  However for Rush Stack, a really great application of this feature would be for the online API reference to contain **hyperlinks to the GitHub source code** for our projects.  That seems pretty straightforward to implement in API Documenter, but it needs some extra information so that we can expand `src/api/Extractor.ts` into a working URL such as `https://github.com/microsoft/rushstack/blob/main/apps/api-extractor/src/api/Extractor.ts`.

Also, the name `sourceFile` is maybe not descriptive enough.  The individual fields we need are more like:

- `projectFolderUrl`: `https://github.com/microsoft/rushstack/tree/main/apps/api-extractor`
- `fileUrlPath`: `src/api/Extractor.ts`
- `fileUrl`: `https://github.com/microsoft/rushstack/blob/main/apps/api-extractor/src/api/Extractor.ts`

One solution would be to make these properties of a class `sourceLocation: SourceLocation`, which can compute `sourceLocation.fileUrl` by appending `projectFolderUrl`+`fileUrlPath`.  This might also provide a container for other related future features, for example what if my filename has special characters and I need to obtain the filename `src/api/Äpfel.ts` separately from its URL `src/api/%C3%84pfel.ts`?

Reply:
(And just to be clear, if we make a `SourceLocation` class it would be serialized directly. It might collect `projectFolderUrl` from the `ApiPackage`, and inherit `fileUrlPath` from a parent API item, and then `fileUrl` might be calculated by appending those two properties.)

Reply:
I took a stab at implementing what you've described above. Take a look and LMK what you think!